iothub/device/devdoc/Convention-based operations.md

```diff
@@ -109,20 +124,25 @@ public Task SubscribeToWritablePropertiesEventAsync(Func<ClientPropertyCollectio
 
 ```csharp
 public class ClientProperties : ClientPropertyCollection {
+    public ClientProperties();
     public ClientPropertyCollection Writable { get; private set; }
 }
 
 public class ClientPropertyCollection : PayloadCollection {
     public ClientPropertyCollection();
     public long Version { get; protected set; }
-    public void Add(IDictionary<string, object> properties, string componentName = null);
+    public void Add(IDictionary<string, object> properties);
```

Comment:
Is this new change to make it similar to other languages? I thought in C# we try not to have too many overloads.

Reply:
No, the idea here was to keep the component name as the first param, for all APIs that do accept a component name. 
This will ensure that we have the same experience throughout; eg. if we do `TryGetValue(string componentName, string propertyName, out T value)` , then we should be able to do `Add(string componentName, string propertyName, object value)`.

The other alternative was to put componentName as the last param for all APIs (since it is optional), but calling this did not feel intuitive:
```csharp
var value = properties.TryGetValue("property1", out T properytValue, "component1");
```

Reply:
This does bring in additional overloads, which we do try to avoid; however, in this case adding optional params gave rise to a non-intuitive API signature.

Reply:
It might be that we have competing priorities, in which case we should discuss which way we'd rather go. Either component name is always first, or it is optional and at the end.

Reply:
Any thoughts about this, Abhipsa?

Reply:
Good point, let us discuss this during PL tomorrow.